pkg/test/framework/components/istio/ingress.go

```diff
@@ -196,22 +217,22 @@ func (c *ingressImpl) callEcho(opts echo.CallOptions) (echo.CallResult, error) {
 		// Default port based on protocol
 		switch s {
 		case scheme.HTTP:
-			addr, port = c.HTTPAddress()
+			addrs, ports = c.HTTPAddresses()
 		case scheme.HTTPS:
-			addr, port = c.HTTPSAddress()
+			addrs, ports = c.HTTPSAddresses()
 		case scheme.TCP:
-			addr, port = c.TCPAddress()
+			addrs, ports = c.TCPAddresses()
 		default:
 			return echo.CallResult{}, fmt.Errorf("ingress: scheme %v not supported. Options: %v+", s, opts)
 		}
 	} else {
-		addr, port = c.AddressForPort(opts.Port.ServicePort)
+		addrs, ports = c.AddressesForPort(opts.Port.ServicePort)
 	}
-
-	if addr == "" || port == 0 {
+	if addrs == nil || ports == nil {
 		scopes.Framework.Warnf("failed to get host and port for %s/%d", opts.Port.Protocol, opts.Port.ServicePort)
 	}
-
+	addr = addrs[0]
+	port = ports[0]
```

Comment:
by default callEcho sets the address to the first remote address that is found, providing two addresses to echo is not possible.
Given that providing two addresses to echo is not possible, my idea was to have "one call" -> "one request for every address found" but that has also proven to be too disruptive of a change (and also a tad opaque).
I plan to revisit this change later to see if there is anything less disruptive that can be done about it